src/Text/Pandoc/Readers/Markdown.hs

```diff
@@ -1741,6 +1742,41 @@ source = do
 linkTitle :: MarkdownParser String
 linkTitle = quotedTitle '"' <|> quotedTitle '\''
 
+wikilinkstuff :: MarkdownParser (F Inlines, String)
+wikilinkstuff = do
+  string "[["
+  (_, raw) <- withRaw $ charsInBalancedBrackets 2
+  guard $ not $ null raw
+  let deunder c = if c == '_' then ' ' else c
+      deiki s = case elemIndices '/' s of
+                  [] -> map deunder s
+                  xs -> map deunder (drop ((last xs) + 1) s)
+  case elemIndex '|' raw of
+    Nothing -> do
+      let rawstr = ((reverse . (drop 2) . reverse) raw)
```

Comment:
Most parens here are superfluous.  The outmost pair doesn't add anything, so this becomes

``` haskell
rawstr = (reverse . (drop 2) . reverse) raw
```

Function application has higher precedence than any infix operator, including `(.)`, so the parens around `(drop 2)` can be dropped as well.

``` haskell
rawstr = (reverse . drop 2 . reverse) raw
```

Now finally, one could use the `($)` operator to eliminate the remaining parens:

``` haskell
rawstr = reverse . drop 2 . reverse $ raw
```

That's a matter of taste, though.